tests/final_chain_test.cpp

```diff
@@ -600,6 +605,26 @@ TEST_F(FinalChainTest, incorrect_estimation_regress) {
   }
 }
 
+TEST_F(FinalChainTest, fee_rewards_distribution) {
```

Comment:
I left this functionality there for now, because because we need to decide what we are doing with this and this is just refactor PR that I was talking about